src/index.js

```diff
@@ -111,26 +238,17 @@ class TemplateContentScript extends ContentScript {
       '.detail-facture'
     )
     const billsDone = await this.runInWorker('getBills')
-    if (billsDone) {
-      await this.saveBills(this.store.allDocuments, {
-        context,
-        fileIdAttributes: ['vendorRef', 'filename'],
-        contentType: 'application/pdf',
-        qualificationLabel: 'energy_invoice'
-      })
-      await this.clickAndWait(
-        'a[href="/clients/mon-compte/mon-contrat"]',
-        '.cadre2'
-      )
-      await this.runInWorkerUntilTrue({ method: 'checkContractPageTitle' })
-      await this.runInWorker('getContract')
-      await this.saveFiles(this.store.contract, {
-        context,
-        fileIdAttributes: ['filename'],
-        contentType: 'application/pdf',
-        qualificationLabel: 'energy_contract'
-      })
-    }
+    return billsDone
+  }
+
+  async fetchContracts() {
+    this.log('info', 'fetchContracts starts')
+    await this.clickAndWait(
+      'a[href="/clients/mon-compte/mon-contrat"]',
+      '.cadre2'
+    )
+    await this.runInWorkerUntilTrue({ method: 'checkContractPageTitle' })
```

Comment:
Didn't we state for a previous konnector that the method called by runInWorkerUntilTrue should use waitFor itself or else we would have instabilities ?